addon-test-support/setup-middleware-reporter.ts

```diff
@@ -1,36 +1,70 @@
 import QUnit from 'qunit';
-import { getContext, getTestMetadata } from '@ember/test-helpers';
-import { AxeResults } from 'axe-core';
+import {
+  currentRouteName,
+  currentURL,
+  getContext,
+  getTestMetadata,
+} from '@ember/test-helpers';
+import { AxeResults, Result } from 'axe-core';
 import { setCustomReporter } from './reporter';
 import { DEBUG } from '@glimmer/env';
 
 interface AxeTestResult {
   moduleName: string;
   testName: string;
-  helperName: string;
+  urls: string[];
+  routes: string[];
+  helpers: string[];
   stack: string;
-  axeResults: AxeResults;
+  violations: Result[];
 }
 
 export const TEST_SUITE_RESULTS: AxeTestResult[] = [];
 
-export function buildResult(axeResults: AxeResults) {
-  let { module, testName } = QUnit.config.current;
-  let testMetaData = getTestMetadata(getContext());
+let currentTestResult: AxeTestResult | undefined = undefined;
+let currentViolationsMap = new Map<string, Result>();
 
-  let stack = (!DEBUG && new Error().stack) || '';
+export async function middlewareReporter(axeResults: AxeResults) {
+  if (axeResults.violations.length === 0) {
+    return;
+  }
+
+  if (!currentTestResult) {
+    let { module, testName } = QUnit.config.current;
+    let testMetaData = getTestMetadata(getContext());
+
+    let stack = (!DEBUG && new Error().stack) || '';
+
+    currentTestResult = {
+      moduleName: module.name,
+      testName,
+      urls: [currentURL()],
+      routes: [currentRouteName()],
+      helpers: testMetaData.usedHelpers,
+      stack,
+      violations: [],
+    };
+  }
```

Comment:
```suggestion
  if (!currentTestResult) {
    let { module, testName } = QUnit.config.current;
    let testMetaData = getTestMetadata(getContext());

    let stack = (!DEBUG && new Error().stack) || '';

    currentTestResult = {
      moduleName: module.name,
      testName,
      urls: new Set(),
      routes: new Set(),
      helpers: testMetaData.usedHelpers,
      stack,
      violations: [],
    };
  }

  currentTestResult.urls.add(currentURL());
  currentTestResult.routes.add(currentRouteName());
```

Reply:
Are you gonna mess with my formatting again, @rwjblue??? :D